stats/opentelemetry/internal/tracing/custom_carrier.go

```diff
@@ -26,16 +26,28 @@ import (
 	"google.golang.org/grpc/metadata"
 )
 
+// PropagationDirection specifies whether the propagation is incoming or
+// outgoing.
+type PropagationDirection int
+
+const (
+	Incoming PropagationDirection = iota // Incoming propagation direction
+	Outgoing                             // Outgoing propagation direction
+)
+
 // CustomCarrier is a TextMapCarrier that uses `context.Context` to store and
-// retrieve any propagated key-value pairs in text format.
+// retrieve any propagated key-value pairs in text format. The propagation
+// direction (incoming or outgoing) determines which keys should the `Keys()`
+// method returns.
 type CustomCarrier struct {
```

Comment:
"Custom" in the name seems to add nothing of value.  Just Carrier is fine since it's in an internal tracing package?

Reply:
yeah since its inside internal `tracing` package, only carrier is fine